gateway/src/resty/resolver/http.lua

```diff
@@ -52,4 +55,23 @@ function _M.connect(self, host, port, ...)
   return ok, err
 end
 
+function _M:set_keepalive()
+  if keepalive_request then
+    local count, err = resty_http.get_reused_times(self)
+    if err then
+      return nil, err
+    end
+    if count >= tonumber(keepalive_request) then
+      resty_http.close(self)
```

Comment:
This line was not in the beginning. What difference does it make?

Reply:
hey, this is part of my comment above:

```
Just a question re closing connections. Do we want to do this manually or as An explained to me, rely on the lua GC?
```

I'm testing with close() but see no differences really, it was more about being explicit about closing connections that we are opening rather than depending on garbage collector. From the test so far, I see no difference

Reply:
Close the connection explicitly instead of letting GC handle it. This address Michal concern above

> Just a question re closing connections. Do we want to do this manually or as An explained to me, rely on the lua GC?

Reply:
understood, thanks.